File: implementations/flutter/design_patterns_flutter/lib/patterns/behavioral/visitor/element_book.dart
import 'element.dart';
import 'visitor.dart';

class Book implements Element {
  final String title;
  final double unitPrice;
  int qty;
  final bool isImported;

  Book(this.title, this.unitPrice, this.qty, [this.isImported = false]);

  @override
  void accept(Visitor v) => v.visitBook(this);
}

File: implementations/react-native/design-patterns-rn/src/patterns/behavioral/visitor/element-book.ts
import type { Visitor } from './visitor';
import type { Element } from './element';

export class Book implements Element {
  public title: string;
  public unitPrice: number;
  public qty: number;
  public isImported: boolean;

  constructor(title: string, unitPrice: number, qty: number, isImported = false) {
    this.title = title;
    this.unitPrice = unitPrice;
    this.qty = qty;
    this.isImported = isImported;
  }

  accept(v: Visitor) { v.visitBook(this); }
}
